tests/testthat/test-integration_metrics.R

```diff
@@ -0,0 +1,109 @@
+# generate testing data
+set.seed(1665)
+merged_sce <- scpcaTools:::sim_sce(n_cells = 300, n_genes = 100, n_empty = 0)
+batches <- rep(c("a", "b", "c"), each = 100)
+barcodes <- rownames(colData(merged_sce))
+
+# set up colData rownames as batch-barcode
+new_rownames <- glue::glue("{batches}-{barcodes}")
+rownames(colData(merged_sce)) <- new_rownames
+
+# Add "sample" to colData
+colData(merged_sce)$sample <- batches
+
+# add PCs for testing (exact numbers don't matter)
+# make a 300x100 matrix
+reducedDim(merged_sce, "PCA") <- matrix(runif(300 * 100, min = 0, max = 100), nrow = 300)
+
+# grab pcs and batches
+pcs <- reducedDim(merged_sce, "PCA")
+batches <- merged_sce$sample
+
+test_that("`set_pc_rownames` works as expected", {
+
+  # expect rownames of returned pcs to be the same as the batch column
+  named_pcs <- set_pc_rownames(pcs, batches)
+  expect_true(all.equal(rownames(named_pcs), batches))
+
+})
+
+test_that("`set_pc_rownames` removes NAs as expected", {
+
+  batches[1:5] <- NA # set a subset of batch labels to NAs
+
+  # expect rownames to be equal to only the batches without NA
+  named_pcs <- set_pc_rownames(pcs, batches)
+  expect_true(all.equal(rownames(named_pcs), batches[!is.na(batches)]))
+  expect_false(length(rownames(named_pcs)) == length(batches))
+})
+
+test_that("`set_pc_rownames` fails without batch labels", {
+
+  short_batches <- c("a", "b", "c")
+  expect_error(scpcaTools:::set_pc_rownames(pcs, short_batches))
+
+})
+
+test_that("`downsample_pcs` works as expected", {
+
+  downsampled <- downsample_pcs(pcs, frac_cells = 0.8)
+  downsampled_pcs <- downsampled$pcs
+  downsampled_batch_labels <- downsampled$batch_labels
+
+  # check that the correct number of cells were downsampled
+  expect_true(length(rownames(downsampled_pcs)) == 0.8*length(rownames(pcs)))
+  expect_true(length(downsampled_batch_labels) == 0.8*length(batches))
```

Comment:
Since sampling does not allow fractional samples...

We might want to use a test matrix with a non-multiple-of-5 number of rows to make this aspect of the test more applicable.
```suggestion
  expect_true(length(rownames(downsampled_pcs)) == floor(0.8*length(rownames(pcs))))
  expect_true(length(downsampled_batch_labels) == floor(0.8*length(batches)))
```

Reply:
I think this is actually okay, because we have 300 cells and downsampling to 80% of those should give you 240 cells? Unless I'm missing something.

Reply:
Yes, in this case it is fine, but I think we might want to check that it works as expected with something like 303 cells.

Reply:
Ah I mis-read non-multiple of 5 as multiple of 5. That makes sense.